packages/util/src/signature.ts

```diff
@@ -44,6 +44,11 @@ export function ecsign(msgHash: Buffer, privateKey: Buffer, chainId: any): any {
   return { r, s, v }
 }
 
+export function ecsignRaw(msgHash: Buffer, privateKey: Buffer): ECDSASignature {
+  const { r, s, v } = ecsign(msgHash, privateKey)
```

Comment:
Small question here, would it be cleaner to make `ecsign` use `ecsignRaw` and then add 27 to it, instead of that we call into `ecsign` and then subtract 27 here? :thinking:

Reply:
I am very much on shaky ground here, but on first look: wouldn't it be cleanest to directly add a `raw` parameter to the `ecsign` method, and then just keep the 27 out completely? 🤔

I know the function signature doesn't get any better with this with this optional `chainId` parameter. But this might generally need some clean-up in the next breaking release, e.g. by using a dict for options like chainId or raw or generally rethink the overall API and naming a bit.

Also, naming: is `raw` the best name for this? Just openly throwing in the ring.

Reply:
I have created a new method here for the reason you note here; otherwise you have to call `ecsign(msg, key, undefined, true)`. But, what we could do, is to change the chainId field to a different name and then allow the values `undefined`, `number`, `BNLike` and `boolean`; in case its a boolean, check if the boolean is true and if that's the case then return the raw signed message.

Have no strong opinion here, waiting for other opinions.

Reply:
Hmm, my fear is a bit that this won't get any better to understand how to use with another method on this.

Let's please not merge this too hastily and keep a bit in discussion how to structure and name things.

Reply:
Idea from @holgerd77: change the `chainId` optional value to an optional dictionary where we can pass in either a `chainId` field or a `raw` boolean.